### PR TITLE
changed some api routes and fixed sign in bug

### PR DESCRIPTION
**Pull request by mihaimazareanu**

### CODE REVIEW

File: pages/accounts.js [hunk 1]
```diff
@@ -12,10 +12,11 @@ import Layout from "../components/Layout";
 import styled from "styled-components";
 
 export default function Accounts() {
-  const {user} = useContext(UserContext);
+  const {user, setUser} = useContext(UserContext);
```
Comment (ufelgen) [inline pages/accounts.js:23]: I'm not familiar with this hook, so I read up on it and as I understand it, it's an easier way to pass states to nested components. You seem to be using it a little differently from the [documentation I found](https://www.w3schools.com/react/react_usecontext.asp) but I guess it works? so who am I to judge ;)

File: pages/accounts.js [hunk 2]
```diff
@@ -78,6 +82,98 @@ export default function Accounts() {
...
+          }
+        } catch {
+          throw new Error(`Fetch failed`);
+        }
+      };
+      getUser();
+    }
+
+    const calculateCurrentAmount = () => {
+      let addAmount = Number(piggyBank?.startAmount);
```
Comment (ufelgen) [inline pages/accounts.js]: I would call the constant "newAmount" since it's possible to both deposit and withdraw

File: pages/accounts.js [hunk 3]
```diff
@@ -78,6 +82,98 @@ export default function Accounts() {
...
+      piggyBank?.transactions?.forEach(transaction => {
+        console.table(transaction);
```
Comment (ufelgen) [inline pages/accounts.js]: do you need this here or can you remove it?

File: pages/accounts.js [hunk 4]
```diff
@@ -78,6 +82,98 @@ export default function Accounts() {
...
+        if (transaction.typeOfTransaction === "deposit") {
+          addAmount = addAmount + Number(transaction.amount);
+          return;
+        } else if (transaction.typeOfTransaction === "withdrawal") {
+          addAmount = addAmount - Number(transaction.amount);
+          return;
+        } else {
+          console.log("Type of transaction is missing");
```
Comment (ufelgen) [inline pages/accounts.js]: could you maybe change this to an alert to let the user know what's wrong, too?

File: pages/accounts.js [hunk 5]
```diff
@@ -78,6 +82,98 @@ export default function Accounts() {
...
+          return;
+        }
+      });
+
+      console.log(typeof addAmount);
```
Comment (ufelgen) [inline pages/accounts.js]: please remove console log

File: pages/api/children/[childId]/accounts/[accountId]/transactions.js [hunk 6]
```diff
@@ -0,0 +1,47 @@
+import connectDB from "../../../../_db/connect-db";
+import {Child} from "../../../../_db/models/Child";
+
+async function handler(req, res) {
+  switch (req.method) {
+    // case "GET":
+    //   try {
+    //     const child = await Child.findById(req.query.childId);
+    //     console.log(req.query.childId);
```
Comment (ufelgen) [inline pages/api/children/[childId]/accounts/[accountId]/transactions.js]: please remove console log

File: pages/api/children/[childId]/accounts/[accountId]/transactions.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+    //       return res.status(500).json({error: "user not found"});
+    //     }
+    //   } catch {
+    //     console.log("error");
+    //   }
+    //   break;
+    case "POST":
+      try {
+        const child = await Child.findById(req.query.childId);
+        // console.log(child);
```
Comment (ufelgen) [inline pages/api/children/[childId]/accounts/[accountId]/transactions.js]: please remove console log

File: pages/api/children/[childId]/accounts/[accountId]/transactions.js [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+          child.accounts
+            .find(
+              account =>
+                JSON.stringify(account._id) ===
+                JSON.stringify(req.body.accountId)
+            )
+            .transactions.push(req.body.transaction);
+        } else {
+          console.log("error");
+          //   child.accounts = [req.body];
```
Comment (ufelgen) [inline pages/api/children/[childId]/accounts/[accountId]/transactions.js]: can this line be deleted?

File: pages/api/children/[childId]/accounts/[accountId]/transactions.js [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+    //     console.log(req.query.childId);
+    //     if (child) {
+    //       return res.status(200).json(child.accounts);
+    //     } else {
+    //       return res.status(500).json({error: "user not found"});
+    //     }
+    //   } catch {
+    //     console.log("error");
+    //   }
+    //   break;
```
Comment (ufelgen) [inline pages/api/children/[childId]/accounts/[accountId]/transactions.js]: is this intentionally commented out?

**Review by ufelgen — COMMENTED** [review]
Hi Mihai! 
disclaimer first: I'm not good with backend issues. I thought I could learn some things from your PR and I did :) 
I made some comments, maybe they're helpful.

File: pages/accounts.js (hunk 1 above)
Comment (mihaimazareanu) [reply pages/accounts.js:23]: Yes, contexts are part of the Global States session that we had and they are really useful indeed.

File: pages/api/children/[childId]/accounts/[accountId]/transactions.js (hunk 9 above)
Comment (mihaimazareanu) [reply pages/api/children/[childId]/accounts/[accountId]/transactions.js]: Yes, I haven't worked on the GET method yet, that's the next step.

**Pull request merged**